frog/server/runDataflow.js

```diff
@@ -54,6 +53,8 @@ const runDataflow = (
     return;
   }
 
+  nodeTypes[type].update(nodeId, { $set: { state: 'computing' } });
```

Comment:
It was very hard for me to understand why moving this line after the if statement helps. You never use the fact that state will be equal to 'computing'. As of now we could remove this line and everything would work the same. 

I thought the problem was due to concurrency issue of code running at the same time, but actually it was only that the state='computed' was being overwritten. 

Will we ever have issue or concurrency? 
If yes, I would suggest replacing the above if statement by `if (node.state === 'computed' || node.state ==='computing')...`